tests/utils/s3_bulk_renaming_test.py

```diff
@@ -0,0 +1,146 @@
+from unittest import mock
+
+import pytest
+
+from utils.s3.s3_bulk_renaming import (
+    detect_timestamps_without_minutes,
+    filter_objects,
+    list_objects,
+    rename_objects,
+    update_timestamp,
+)
+
+
+@pytest.fixture
+def sample_objects():
+    return [
+        {"Key": "file1.txt"},
+        {"Key": "file2.jpg"},
+        {"Key": "file3.txt"},
+        {"Key": "file4.txt"},
+        {
+            "Key": (
+                "diagnostics.zarr/RTMA/WCOSS/CONUS/RRFS/REALTIME/uv/"
+                "2023-06-06T16/ges/observation/1.1"
+            )
+        },
+        {
+            "Key": (
+                "diagnostics.zarr/RTMA/WCOSS/CONUS/RRFS/REALTIME/uv/"
+                "2023-06-06T16:00/ges/observation/1.1"
+            )
+        },
+    ]
+
+
+class TestDetectTimestamps:
+    def test_without_minutes(self):
+        key = (
+            "diagnostics.zarr/RTMA/WCOSS/CONUS/RRFS/REALTIME/uv/"
+            "2023-06-06T16/ges/observation/1.1"
+        )
+
+        assert key == detect_timestamps_without_minutes(key)
+
+    def test_with_minutes(self):
+        key = (
+            "diagnostics.zarr/RTMA/WCOSS/CONUS/RRFS/REALTIME/uv/"
+            "2023-06-06T16:00/ges/observation/1.1"
+        )
+
+        assert detect_timestamps_without_minutes(key) is None
+
+    def test_no_timestamp(self):
+        key = "foo/bar/baz"
+
+        assert detect_timestamps_without_minutes(key) is None
```

Comment:
Good catch. I was wondering why that function tested for both timestamp formats with and without minutes.